### DISCUSSION TITLE
Github-Desktop Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by wjberey** [community/community#58272] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Only external editor in drop-down under Options is GEdit.  VS Code is installed on my system.  How do I add VS Code to the external editor dropdown?  Thanks!

**Comment by the-officialjosh** [comment]
To add Visual Studio Code (VS Code) as an external editor in GitHub Desktop on Ubuntu, you can follow these steps:

1. If you don't see VS Code listed in the drop-down menu, click on the "Custom" option at the bottom of the list.
2. In the "Choose Application" dialogue, navigate to the location where VS Code is installed on your system. The default installation path is usually `/usr/bin/code`. If you installed VS Code using a different method, you might need to locate the appropriate path.
3. Select the `code` executable file and click "Open" to add VS Code as the external editor.

Now, when you want to open a file in an external editor from GitHub Desktop, you should be able to select VS Code from the drop-down menu.

**Reply by wjberey** [reply]
Thanks but there is no custom either.  The only item in the dropdown is Gedit.  Any other suggestions?  Thanks!

**Comment by aahouzi** [comment]
To add VS Code as an option in the external editor dropdown in GitHub, you'll need to configure it in your system's settings. Here's how you can do it:

1. Open VS Code and go to the Command Palette. You can do this by pressing Ctrl+Shift+P (Windows/Linux) or Cmd+Shift+P (Mac).
2. Search for "Shell Command: Install 'code' command in PATH" and select it from the list. This command sets up VS Code to be executable from the command line.
3. Once the command has executed successfully, you should be able to run code from the command line or terminal.

Now that the code command is available in your system, you can configure GitHub to use VS Code as the external editor:

1. Open GitHub in your web browser and log in to your account.
2. Click on your profile picture in the top-right corner, then select "Settings" from the dropdown menu.
3. In the left sidebar, click on "Developer settings."
4. Scroll down to the "Editor" section and select the "External Editor" option.
5. In the "Editor dropdown" section, click on the dropdown menu and select "Custom."
6. In the "Editor command" field, enter code --wait. This command tells GitHub to use VS Code as the external editor and --wait ensures that the editor stays open until you close the file.
7. Click "Save" to apply the changes.

After following these steps, you should see "VS Code" as an option in the external editor dropdown when editing files in GitHub.

Please note that these instructions assume you have VS Code installed and configured correctly on your system. If you encounter any issues during the process, consult the official VS Code documentation or seek further assistance from their support channels.

**Reply by wjberey** [reply]
Thanks but:
Shell Command: Install 'code' command in PATH not an option in VS Code - do I need an extension?
"Editor" not a section under Developer settings - is it somewhere else maybe?

**Comment by sepehr-sanaei** [comment]
> ### Select Topic Area
> 
> Question
> ### Body
> 
> Only external editor in drop-down under Options is GEdit. VS Code is installed on my system. How do I add VS Code to the external editor dropdown? Thanks!



> ### Select Topic Area
> 
> Question
> ### Body
> 
> Only external editor in drop-down under Options is GEdit. VS Code is installed on my system. How do I add VS Code to the external editor dropdown? Thanks!

HI, I've had the same problem and you can add vscode by following these steps:
1.click on file on top left corner
2.options
3.integrations
In here add Visual Studio Code in External editor and click on save
Hope it helps you
